### PR TITLE
[serve] Add replica pid to replica startup logs

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Log the replica PID when it starts. This will help with debugging. Sometimes we might want to check Ray core logs, and PID can be used to cross reference replica ID with actor logs.

Example:
```
INFO 2024-12-11 20:57:19,397 controller 62696 -- Replica(id='a1mjqmdq', deployment='api', app='default') started successfully on node '7acc041e50ba72d78dbf376298ddd1f1975e0f7b7e9e712cd9ac99f7' after 0.6s (PID: 62691). Replica constructor, reconfigure method, and initial health check took 0.0s.
```

### CODE REVIEW

File: python/ray/serve/_private/deployment_state.py [hunk 1]
```diff
@@ -2000,7 +2005,7 @@ def _check_startup_replicas(
                 replica_startup_message = (
                     f"{replica.replica_id} started successfully "
                     f"on node '{replica.actor_node_id}' after "
-                    f"{e2e_replica_start_latency:.1f}s."
+                    f"{e2e_replica_start_latency:.1f}s (PID: {replica.actor_pid})."
                 )
                 if replica.initialization_latency_s is not None:
                     # This condition should always be True. The initialization
```
Comment (GeneDer) [inline python/ray/serve/_private/deployment_state.py:2011]: Should we add it to the message below as well?

**Review by GeneDer — APPROVED** [review]
LGTM!

File: python/ray/serve/_private/deployment_state.py (hunk 1 above)
Comment (zcin) [reply python/ray/serve/_private/deployment_state.py:2011]: I think it is all one log message?

Comment (GeneDer) [reply python/ray/serve/_private/deployment_state.py:2011]: oo I misread it XD

**Pull request merged**